### PR TITLE
[FIX] Message Erasure Type "Keep" Messages not working

### PR DESCRIPTION
**Pull request by arshxyz**
## Proposed changes
Check thread on https://github.com/RocketChat/Rocket.Chat/issues/23896

## Issue(s)
https://github.com/RocketChat/Rocket.Chat/issues/23896

### CODE REVIEW

**Review by murtaza98 — CHANGES_REQUESTED** [review]
Hey @arshxyz I feel the first approach you mentioned [here](https://github.com/RocketChat/Rocket.Chat/issues/23896#issuecomment-988779797) looks more elegant since there is no need to send a broadcast in case the event is not valid. I'd recommend using that solution instead. Thanks!!

**Comment by arshxyz** [comment]
I've made the changes accordingly, thanks

File: server/modules/listeners/listeners.module.ts [hunk 1]
```diff
@@ -59,10 +59,14 @@ export class ListenersModule {
 			});
 		});
 
-		service.onEvent('user.deleted', ({ _id: userId }) => {
-			notifications.notifyLoggedInThisInstance('Users:Deleted', {
-				userId,
-			});
+		service.onEvent('user.deleted', ({ _id: userId }, keepMessages) => {
```
Comment (murtaza98) [inline server/modules/listeners/listeners.module.ts:62]: Do we still need this change?

File: server/sdk/lib/Events.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export type EventSignatures = {
 	'stream'([streamer, eventName, payload]: [string, string, string]): void;
 	'subscription'(data: { action: string; subscription: Partial<ISubscription> }): void;
 	'user.avatarUpdate'(user: Partial<IUser>): void;
-	'user.deleted'(user: Partial<IUser>): void;
+	'user.deleted'(user: Partial<IUser>, keepMessages: boolean): void;
```
Comment (murtaza98) [inline server/sdk/lib/Events.ts:39]: Same goes here

Comment (arshxyz) [reply server/sdk/lib/Events.ts]: nope, both changes have been removed

Comment (arshxyz) [reply server/sdk/lib/Events.ts]: should be fine now

**Review by murtaza98 — APPROVED** [review]

**Pull request merged**